address_lookup_tables/src/address_lookup_table_store.rs

```diff
@@ -0,0 +1,218 @@
+                Some(alt_data) => {
+                    let size = alt_data.len() as u8;
+                    if alt.readonly_indexes.iter().any(|x| *x >= size)
+                        || alt.writable_indexes.iter().any(|x| *x >= size)
+                    {
+                        Some(alt.account_key)
+                    } else {
+                        None
+                    }
+                }
+                None => Some(alt.account_key),
+            })
+            .collect_vec();
+        self.load_alts_list(&accounts_to_load).await;
+    }
+
+    pub async fn reload_alt_account(&self, address: &Pubkey) {
+        log::info!("Reloading {address:?}");
+
+        let account = match self
+            .rpc_client
+            .get_account_with_commitment(address, CommitmentConfig::processed())
+            .await
+        {
+            Ok(acc) => acc.value,
+            Err(e) => {
+                log::error!(
+                    "Error for fetching address lookup table {} error :{}",
+                    address.to_string(),
+                    e.to_string()
+                );
+                None
+            }
+        };
+        match account {
+            Some(account) => {
+                self.save_account(address, &account.data);
+            }
+            None => {
+                log::error!("Cannot find address lookup table {}", address.to_string());
+            }
+        }
+    }
+
+    async fn load_accounts(&self, alt: &Pubkey, accounts: &[u8]) -> Option<Vec<Pubkey>> {
+        let alt_account = self.map.get(alt);
+        match alt_account {
+            Some(alt_account) => Some(
+                accounts
+                    .iter()
+                    .map(|i| alt_account[*i as usize])
+                    .collect_vec(),
+            ),
+            None => {
+                log::error!("address lookup table {} was not found", alt);
+                None
+            }
+        }
+    }
+
+    pub async fn get_accounts(&self, alt: &Pubkey, accounts: &[u8]) -> Vec<Pubkey> {
+        match self.load_accounts(alt, accounts).await {
+            Some(x) => x,
+            None => {
+                // forget alt for now, start loading it for next blocks
+                // loading should be on its way
+                vec![]
+            }
+        }
+    }
+
+    pub fn serialize_binary(&self) -> Vec<u8> {
+        bincode::serialize::<BinaryALTData>(&BinaryALTData::new(&self.map)).unwrap()
+    }
+
+    pub fn load_binary(&self, binary_data: Vec<u8>) {
+        let binary_alt_data = bincode::deserialize::<BinaryALTData>(&binary_data).unwrap();
+        for (alt, accounts) in binary_alt_data.data.iter() {
+            self.map.insert(*alt, accounts.clone());
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct BinaryALTData {
+    pub data: Vec<(Pubkey, Vec<Pubkey>)>,
+}
+
+impl BinaryALTData {
+    pub fn new(map: &Arc<DashMap<Pubkey, Vec<Pubkey>>>) -> Self {
```

Comment:
nit: would not call that _new_. it's an intermediate mapper function. could be map_from() or so

Reply:
There is no method like that, dashmap iterator is a special kind of iterator.